rfc100-resource-lifecycle.md

```diff
@@ -89,6 +89,62 @@ Factors that influence and inform the decision to deprecate a resource include:
 * Resource is added to core chef and documentation is updated.
 * Add deprecated cookbook warnings for conflicting cookbooks.
 
+#### Timeline of resource migration
+
+While we strive to ensure the migration of a resource from a cookbook to core is
+100% compatible, small issues can still arise. In order to ensure that Chef feature
+releases do not cause destabilization, we follow a multi-part timeline for the
+migration. Starting from the top:
+
+1. A resource is added in a cookbook.
+2. That resource is nominated for core inclusion (see above section).
+3. The resource is added to core with an annotation of `allow_cookbook_override: true`.
+4. Chef releases a minor (features-only) version with the new resource. It will
+   remain inert if the original cookbook is active until the given version.
+5. At some point in the future, the cookbook does the major version release. Users
+   can upgrade to it when they feel ready, otherwise they will see no change in
+   behavior even with the new Chef release.
+6. The following April, as Chef prepares for the yearly major release, all pending
+   `allow_cookbook_override` annotations will be removed.
```

Comment:
An alternative here would be to use a Chef version matcher as the value instead of `true` so it can turn off automatically. But this should always (famous last words) be the next major version following when the resource is added to core, which is already documented in the `introduced` method/annotation. How "always" do we think that will be?

Reply:
I like the idea of the version matcher... 

Though shouldn't those get cleaned up eventually anyway?